source/common/upstream/eds.cc

```diff
@@ -22,6 +22,8 @@ EdsClusterImpl::EdsClusterImpl(
     Stats::ScopePtr&& stats_scope, bool added_via_api)
     : BaseDynamicClusterImpl(cluster, runtime, factory_context, std::move(stats_scope),
                              added_via_api),
+      Envoy::Config::SubscriptionBase<envoy::config::endpoint::v3::ClusterLoadAssignment>(
```

Comment:
FWIW, we're now slowly drifting into the space of true multiple inheritance, which is used somewhat sparingly in Envoy. Not sure if the horse has fully bolted here, @mattklein123?

Reply:
TBH I don't really have a strong opinion on this. I guess I'm of an opinion that our default should be no, but if there is a good case for it I don't see any major issue. cc @envoyproxy/maintainers